pkg/controller/operators/catalog/operator.go

```diff
@@ -1424,49 +1426,52 @@ func (o *Operator) syncInstallPlans(obj interface{}) (syncError error) {
 }
 
 type installPlanTransitioner interface {
-	ResolvePlan(*v1alpha1.InstallPlan) error
 	ExecutePlan(*v1alpha1.InstallPlan) error
 }
 
 var _ installPlanTransitioner = &Operator{}
 
-func transitionInstallPlanState(log *logrus.Logger, transitioner installPlanTransitioner, in v1alpha1.InstallPlan, now metav1.Time) (*v1alpha1.InstallPlan, error) {
+func transitionInstallPlanState(log logrus.FieldLogger, transitioner installPlanTransitioner, in v1alpha1.InstallPlan, now metav1.Time, timeout time.Duration) (*v1alpha1.InstallPlan, error) {
 	out := in.DeepCopy()
 
 	switch in.Status.Phase {
 	case v1alpha1.InstallPlanPhaseRequiresApproval:
 		if out.Spec.Approved {
-			log.Debugf("approved, setting to %s", v1alpha1.InstallPlanPhasePlanning)
 			out.Status.Phase = v1alpha1.InstallPlanPhaseInstalling
+			out.Status.Message = ""
+			log.Debugf("approved, setting to %s", out.Status.Phase)
 		} else {
 			log.Debug("not approved, skipping sync")
 		}
 		return out, nil
 
 	case v1alpha1.InstallPlanPhaseInstalling:
+		if out.Status.StartTime == nil {
```

Comment:
Can we use [IsZero()](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/time.go#L58-L64) which checks for nil and zero.
Just thinking if StartTime could ever be 0, which would fail this check?
```suggestion
		if out.Status.StartTime.IsZero() {
```

Reply:
Or I guess we're considering if this field is ever unset which is not quite the same as 0.

Reply:
This should never be modified after it's set the first time, even if it was set to zero. I can image two general scenarios where it might be zero: an admin manually patches it or the system clock is wrong. The result would be that retries wouldn't happen. So it'd be inconvenient, but assigning a new value would either be fighting with an admin or relying on the same broken clock.